include/swift/AST/SearchPathOptions.h

```diff
@@ -509,6 +510,14 @@ class SearchPathOptions {
   using CrossImportMap = llvm::StringMap<std::vector<std::string>>;
   CrossImportMap CrossImportInfo;
 
+  /// CanImport information passed from scanning.
+  struct CanImportInfo {
+    std::string ModuleName;
+    llvm::VersionTuple Version;
+    llvm::VersionTuple UnderlyingVersion;
```

Comment:
I don't understand the modeling here: it looks like you actually get a separate CanImportInfo for version vs. underlying version, so why do we store both instead of a flag?

Reply:
Yes, those are two separate versions, one for swift module and one for clang module, and they can co-exists for one module decl name because swift module can be an overlay for clang module.

Alternative is to module clang and swift separately and use different options in swift-frontend.

Reply:
I am guessing this is for some Swift client that may have both:
```
#if canImport(Foo, _version: 10.0) { ... }
#if canImport(Foo, _underlyingVersion: 10.0) { ... }
```
The names of these version tuples here at least, are confusing (they are confusing in the `canImport` directive itself, but that's a separate point). Maybe `UnderlyingClangModuleVersion`. Also, IIUC these tuples represent the lowest version for this module for which the versioned `canImport` evaluates to `true`, right? That's worth capturing in the comments here as well.

Reply:
No, it is the version of the module discovered by scanner if version ever requested.

That said, there might be a corner case I need to fix. Looking...

Reply:
> Maybe UnderlyingClangModuleVersion

Current variable actually matches the parameter in `canImport` :)